staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -434,6 +439,16 @@ func NewCacherFromConfig(config Config) (*Cacher, error) {
 		)
 	}()
 
+	if utilfeature.DefaultFeatureGate.Enabled(features.ConsistentListFromCache) {
+		supports, err := cacher.Supports(storage.RequestWatchProgress)
```

Comment:
(assuming that you can configure a separate etcd cluster per resource) if this fails due to a transient error during construction then in line 740 we always say the the `requestWatchProgress` is not supported because there is no component that would try to evaluate the feature availability one more time. 

In addition to that this code is sending a request to etcd during construction time which can fail and be delayed. Should we move this logic to a controller or do lazy evaluation during the first call to the `Supports` method ?

Reply:
Addressed in new changes.